src/omnikinverter/models.py

```diff
@@ -2,11 +2,96 @@
 
 from __future__ import annotations
 
+import json
 import re
 from dataclasses import dataclass
 from typing import Any, cast
 
-from .exceptions import OmnikInverterWrongSourceError, OmnikInverterWrongValuesError
+from .exceptions import (
+    OmnikInverterError,
+    OmnikInverterWrongSourceError,
+    OmnikInverterWrongValuesError,
+)
+
+
+@dataclass
+class WebResponse:
+    """Holds an unparsed response from the connected Omnik."""
+
+    response_string: str
+    source_type: str
+
+    def inverter(self) -> Inverter:
+        """Get values from your Omnik Inverter.
+
+        Returns
+        -------
+            An Inverter data object from the Omnik Inverter.
+
+        Raises
+        ------
+            OmnikInverterError: Unknown source type.
+
+        """
+        if self.source_type == "json":
+            return Inverter.from_json(json.loads(self.response_string))
+        if self.source_type == "html":
+            return Inverter.from_html(self.response_string)
+        if self.source_type == "javascript":
+            return Inverter.from_js(self.response_string)
+
+        msg = f"Unknown source type `{self.source_type}`"  # pragma: no cover
+        raise OmnikInverterError(msg)  # pragma: no cover
+
+    def device(self) -> Device:
+        """Get Device values from your Omnik Inverter.
+
+        Returns
+        -------
+            A Device data object from the Omnik Inverter.
+
+        Raises
+        ------
+            OmnikInverterError: Unknown source type.
+
+        """
+        if self.source_type == "json":
+            return Device.from_json(json.loads(self.response_string))
+        if self.source_type == "html":
+            return Device.from_html(self.response_string)
+        if self.source_type == "javascript":
+            return Device.from_js(self.response_string)
+
+        msg = f"Unknown source type `{self.source_type}`"  # pragma: no cover
+        raise OmnikInverterError(msg)  # pragma: no cover
```

Comment:
Previously I had some TODOs here, because both returns should be unreachable, since the object shouldn't be constructed with anything else than `source_type`.

In addition, if it's `json` we only need to parse the `response_string` once.

Is there a way in which we could _either_ store the parsed `json` object, or a string for HTML, or a string for JS, but nothing else (and make that type-safe)?